src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -1530,12 +1530,20 @@ export class NotebookModel extends Disposable implements INotebookModel {
 	public undo(): void {
 		if (this.undoService.canUndo(this.notebookUri)) {
 			this.undoService.undo(this.notebookUri);
+			const azdata_notebook_guid: string = this.getMetaValue('azdata_notebook_guid');
+			this.adstelemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.UndoCell)
```

Comment:
You don't want to capture what type of undo/redo operation we're doing as well? That seems like it could be useful as well.

Reply:
Yeah, I was thinking about that but the undoService only gives me access to the last element from the undoStack. Let me try add it in the IResourceElement instead.